src/hooks/use-previous.ts

```diff
@@ -0,0 +1,12 @@
+import { useRef, useEffect } from 'react'
+
+/**
+ * Keep track of the previous value of a variable
+ */
+export default function usePrevious<T = never>(value: T) {
```

Comment:
Swiped this from the existing `hashicorp/learn` codebase - in general the toast logic implementation in this PR is pretty similar. Used to detect when progress status for a tutorial changes.